hoomd/write/gsd.py

```diff
@@ -141,6 +141,9 @@ class GSD(Writer):
         truncate (bool): When `True`, truncate the file and write a new frame 0
             each time this operation triggers.
         dynamic (list[str]): Quantity categories to save in every frame.
+        write_diameter (bool): When `False`, do not write
```

Comment:
Should this be part of dynamic? It sort of fulfills the same role except being only a single property.

Reply:
I made it separate because it isn't the same as dynamic. If `diameter` was outside the dynamic list, then it would still be written on frame 0. I would like to avoid writing it at all unless the user opts in.

Reply:
Refactoring complete. As a bonus, the more efficient memory accesses boost performance by a few % in the LJ hoomd benchmark: 3030 TPS to 3130.